src/libos/crates/host-socket/src/stream/states/connected/recv.rs

```diff
@@ -62,7 +62,28 @@ impl<A: Addr + 'static, R: Runtime> ConnectedStream<A, R> {
             }
             let events = self.common.pollee().poll(mask, None);
             if events.is_empty() {
-                poller.as_ref().unwrap().wait().await?;
+                info!("recv timeout = {:?}", self.common.recv_timeout());
+                let ret = poller
+                    .as_ref()
+                    .unwrap()
+                    .wait_timeout(self.common.recv_timeout().as_mut())
+                    .await;
+                info!("ret = {:?}", ret);
+                if let Err(e) = ret {
+                    warn!("recv wait errno = {:?}", e.errno());
+                    match e.errno() {
+                        ETIMEDOUT => {
```

Comment:
I think the case of interruption should be considered.
If the `MSG_WAITALL` flag is included and wait_timeout returns due to an interrupt, should it return  `EINTR` or the size of the received messge ?

Reply:
Done. Thanks for your review.